### ISSUE TITLE
Dfdaemon does not gracefully stop when handling a SIGTERM

### ISSUE THREAD

**Issue by embroede** [dragonflyoss/Dragonfly2#2761]
### Bug report:

In our testing of versions v2.0.9 and v2.1.11, when a download is in progress (using proxy mode), a SIGTERM sent to the dfget/dfdaemon process always results in the download being interrupted (and failing).

### Expected behavior:

We would expect that any in-progress downloads should finish successfully, while no new downloads would be able to start.

### How to reproduce it:

Start dfdaemon, then set proxy variables and start a download. While the download is in progress, run `kill <dfget pid>`.

### Environment:

- Dragonfly version: v2.0.9 and v2.1.11
- OS: Linux
- Kernel (e.g. `uname -a`): 
- Others:

**Comment by embroede** [comment]
Here are some logs for what we see when stopping dfdaemon while a download is in progress:
```json
{"level":"error","ts":"2023-09-19 12:13:34.781","caller":"peer/peertask_conductor.go:827","msg":"receive peer packet failed: rpc error: code = Unavailable desc = error reading from server: read tcp <REDACTED>-><REDACTED>: use of closed network connection","peer":"<REDACTED>","task":"<REDACTED>","component":"PeerTask","trace":"<REDACTED>","stacktrace":"
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).confirmReceivePeerPacketError
    /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:827
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).receivePeerPacket
    /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:707
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).pullPiecesWithP2P
    /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:558
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).pullPieces
    /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:538"}
{"level":"info","ts":"2023-09-19 12:13:34.781","caller":"daemon/daemon.go:841","msg":"dynconfig client closed"}
{"level":"debug","ts":"2023-09-19 12:13:34.781","caller":"peer/peertask_manager.go:426","msg":"delete done task <REDACTED> in running tasks"}
{"level":"info","ts":"2023-09-19 12:13:34.781","caller":"daemon/daemon.go:723","msg":"dynconfig start successfully"}
{"level":"info","ts":"2023-09-19 12:13:34.781","caller":"daemon/daemon.go:848","msg":"manager client closed"}
{"level":"error","ts":"2023-09-19 12:13:34.781","caller":"peer/peertask_conductor.go:1437","msg":"peer task failed, code: 1500, reason: ","peer":"<REDACTED>","task":"<REDACTED>","component":"PeerTask","trace":"<REDACTED>","stacktrace":"
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).fail
    /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:1437
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).cancel.func1
    /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:458
sync.(*Once).doSlow
    /usr/local/go/src/sync/once.go:74
sync.(*Once).Do
    /usr/local/go/src/sync/once.go:65
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).cancel
    /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:455
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).confirmReceivePeerPacketError
    /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:829
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).receivePeerPacket
    /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:707
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).pullPiecesWithP2P
    /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:558
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).pullPieces
    /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:538"}
```

**Comment by embroede** [comment]
@jim3ma  Let me know if there's anything I can do to help

**Comment by embroede** [comment]
This is observed when dfdaemon is proxying a `singularity pull oras://<url>`, which is https traffic. We do this by setting the proxy environment variables to `localhost:<dfdaemon_port>` and then running the `singularity pull`.

**Comment by jim3ma** [comment]
> This is observed when dfdaemon is proxying a `singularity pull oras://<url>`, which is https traffic. We do this by setting the proxy environment variables to `localhost:<dfdaemon_port>` and then running the `singularity pull`.

Okay, I'm working on it.

**Comment by embroede** [comment]
Thanks!

**Comment by embroede** [comment]
@jim3ma @gaius-qi  Thanks for fixing this, we tested it and it is working perfectly. I have two questions:

1. Would you be able to backport this fix to 2.0.9, maybe released as 2.0.10, or 2.0.9-<patch tag or number>
2. 
3. As the releases are now no longer being marked as beta/rc etc., is there a way to tell which is a "stable" release?

Thanks!
Eddie
